.github/workflows/ci.yml

```diff
@@ -582,21 +582,95 @@ jobs:
             bash -euo pipefail -c \
             "./build_tools/cmake/build_riscv.sh && tests/riscv32/smoke.sh"
 
-  riscv64:
-    needs: [should_run, build_all, build_tf_integrations]
+  build_riscv64:
+    needs: [should_run, build_all]
     if: needs.should_run.outputs.should-run == 'true'
     runs-on:
       # Pseudo-ternary hack and order matters. See comment at top of file.
       - self-hosted
       - runner-group=${{ github.event_name == 'pull_request' && 'presubmit' || 'postsubmit' }}
       - cpu
       - os-family=Linux
+    outputs:
+      build-dir: ${{ steps.build.outputs.build-dir }}
+      build-dir-archive: ${{ steps.archive.outputs.build-dir-archive }}
+      build-dir-gcs-artifact: ${{ steps.upload.outputs.build-dir-gcs-artifact }}
+      tools-dir: ${{ steps.build.outputs.tools-dir }}
+      tools-archive: ${{ steps.archive.outputs.tools-archive }}
+      tools-gcs-artifact: ${{ steps.upload.outputs.tools-gcs-artifact }}
     env:
       BUILD_RISCV_DIR: "build-riscv64"
       RISCV_CONFIG: "rv64"
       BUILD_DIR: ${{ needs.build_all.outputs.build-dir }}
       BUILD_DIR_ARCHIVE: ${{ needs.build_all.outputs.build-dir-archive }}
       BUILD_DIR_GCS_ARTIFACT: ${{ needs.build_all.outputs.build-dir-gcs-artifact }}
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+        with:
+          submodules: true
+      - name: "Downloading build dir archive"
+        run: gcloud alpha storage cp "${BUILD_DIR_GCS_ARTIFACT}" "${BUILD_DIR_ARCHIVE}"
+      - name: "Extracting build dir archive"
+        run: tar -xf "${BUILD_DIR_ARCHIVE}"
+      - name: "Cross-compiling riscv64"
+        id: build
+        run: |
+          ./build_tools/github_actions/docker_run.sh \
+            --env "RISCV_CONFIG=${RISCV_CONFIG}" \
+            --env "BUILD_RISCV_DIR=${BUILD_RISCV_DIR}" \
+            --env "IREE_HOST_BINARY_ROOT=${BUILD_DIR}/install" \
+            gcr.io/iree-oss/riscv@sha256:720bc0215d8462ea14352edc22710a6ce4c0c1daff581d179dd173885f1d8a35 \
+            bash -euo pipefail -c "./build_tools/cmake/build_riscv.sh"
+          echo "::set-output name=build-dir::${BUILD_RISCV_DIR}"
+          echo "::set-output name=tools-dir::${BUILD_RISCV_DIR}/tools"
+      - name: "Creating riscv64 cross-compiled build dir and tool archives"
+        id: archive
+        env:
+          BUILD_RISCV_DIR_ARCHIVE: ${{ env.BUILD_RISCV_DIR }}.tar.zst
+          TOOLS_ARCHIVE: riscv64-tools.tar
+        run: |
+          tar --exclude '*.a' --exclude '*.o' \
+            -I 'zstd -T0' \
+            -cf ${BUILD_RISCV_DIR_ARCHIVE} ${BUILD_RISCV_DIR}
+          tar -cf ${TOOLS_ARCHIVE} \
```

Comment:
How about running install and then creating an archive of the whole install directory?

Reply:
Splitted the jobs. So this doesn't apply anymore